### PR TITLE
fix: TT 20 calculate time difference when new entries added

### PR DESCRIPTION
**Pull request by wobravo**
Calculate time difference when new entries added

### CODE REVIEW

File: src/app/modules/shared/components/details-fields/details-fields.component.ts [hunk 1]
```diff
@@ -130,6 +130,17 @@ export class DetailsFieldsComponent implements OnChanges, OnInit {
     this.end_date.setValue($event);
   }
 
+  getTimeDifference() {
```
Comment (thegreatyamori) [inline src/app/modules/shared/components/details-fields/details-fields.component.ts:133]: Can you add a return type to the function?

File: src/app/modules/shared/components/details-fields/details-fields.component.spec.ts [hunk 2]
```diff
@@ -512,6 +512,27 @@ describe('DetailsFieldsComponent', () => {
...
+      case: 'positive should return correctly diff',
+      entryDates: {
+        start_date: '2021-04-15',
+        end_date: '2021-04-15',
+        start_hour: '18:05',
+        end_hour: '19:00',
+      },
+      expectedTimeDiff: '00:55',
+    },
+  ];
```
Comment (thegreatyamori) [inline src/app/modules/shared/components/details-fields/details-fields.component.spec.ts:527]: Here, I'm not see the other test case with negative assert.

**Review by thegreatyamori — APPROVED** [review]
Good job !

File: src/app/modules/shared/components/details-fields/details-fields.component.spec.ts [hunk 3]
```diff
@@ -512,6 +512,39 @@ describe('DetailsFieldsComponent', () => {
...
+      expectedTimeDiff: '00:00',
+    },
+  ];
+  diffParams.map((param) => {
+    it(`if [start_date, start_hour] and [end_date, end_hour] diff is ${param.case}`, () => {
+      component.entryForm.setValue({ ...formValues, ...param.entryDates });
+      const timeDiff = component.getTimeDifference();
+
+      expect(timeDiff).toBe(param.expectedTimeDiff);
+      expect(timeDiff).toBe(param.expectedTimeDiff);
```
Comment (scastillo-jp) [inline src/app/modules/shared/components/details-fields/details-fields.component.spec.ts:544]: Delete this line is duplicate

**Pull request merged**